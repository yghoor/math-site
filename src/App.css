/* Style Pages */

.home-page {
  margin: 12px;
}

.home-page h1 {
  font-family: Arial, Helvetica, sans-serif;
  margin-bottom: 36px;
}

.home-page p {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 16px;
  width: 90%;
}

.math-page {
  display: flex;
  justify-content: center;
}

.math-page h1 {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 36px;
}

.quote-page {
  height: 80vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.quote-page p {
  margin: 0;
  padding: 0;
  width: 75%;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 32px;
}
