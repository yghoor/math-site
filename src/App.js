/* eslint-disable no-unused-vars */
import './App.css';
import Home from './pages/Home.js';
import Calculator from './components/Calculator.js';

function App() {
  return (
    <div className="App">
      <Navbar />
    </div>
  );
}

export default App;

const Navbar = () => (

  <header>
    <nav className="page-navbar">
      <h1>Math Magicians</h1>

      <ul className="nav-links">
        <li>
          <NavLink to="/" activeclassname="active" >Home</NavLink>
        </li>

        <li>
          <NavLink to="/math" activeclassname="active" >Calculator</NavLink>
        </li>

        <li>
          <NavLink to="/quote" activeclassname="active" >Quote</NavLink>
        </li>
      </ul>
    </nav>
  </header>
);
