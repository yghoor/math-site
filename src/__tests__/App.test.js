/* eslint-disable no-unused-vars */
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import App from '../App.js';

describe('App', () => {
  test('renders App', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders "Home" page by default', () => {
    render(<App />);

    const homePageHeader = screen.getByRole('heading', { name: 'Welcome to our page!' });

    expect(homePageHeader).toBeInTheDocument();
  });
});

describe('Navbar component', () => {
  test('renders navbar header', () => {
    render(<App />);

    const navHeader = screen.getByRole('heading', { name: 'Math Magicians' });

    expect(navHeader).toBeInTheDocument();
  });

  test('renders navbar links', () => {
    render(<App />);
    const homeLink = screen.getByRole('link', { name: 'Home' });
    const mathLink = screen.getByRole('link', { name: 'Calculator' });
    const quoteLink = screen.getByRole('link', { name: 'Quote' });

    expect(homeLink).toBeInTheDocument();
    expect(mathLink).toBeInTheDocument();
    expect(quoteLink).toBeInTheDocument();
  });

  test('"Home" link works', () => {
    render(<App />);
    const homeLink = screen.getByRole('link', { name: 'Home' });

    userEvent.click(homeLink);
    const homePageHeader = screen.getByRole('heading', { name: 'Welcome to our page!' });

    expect(homePageHeader).toBeInTheDocument();
  });

  test('"Calculator" link works', () => {
    render(<App />);
    const mathLink = screen.getByRole('link', { name: 'Calculator' });

    userEvent.click(mathLink);
    const calculatorPageHeader = screen.getByRole('heading', { name: 'Let\'s do some math!' });

    expect(calculatorPageHeader).toBeInTheDocument();
  });

  test('"Quote" link works', () => {
    const { container } = render(<App />);
    const quoteLink = screen.getByRole('link', { name: 'Quote' });

    userEvent.click(quoteLink);
    const quotePageDiv = container.getElementsByClassName('quote-page')[0];

    expect(quotePageDiv).toBeInTheDocument();
  });
});