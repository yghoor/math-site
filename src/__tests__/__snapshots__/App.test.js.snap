// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
Array [
  <header>
    <nav
      className="page-navbar"
    >
      <h1>
        Math Magicians
      </h1>
      <ul
        className="nav-links"
      >
        <li>
          <a
            activeclassname="active"
            aria-current="page"
            className="active"
            href="#/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            activeclassname="active"
            className=""
            href="#/math"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            activeclassname="active"
            className=""
            href="#/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="home-page"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet quod fugit incidunt, possimus ea voluptatibus. Error tenetur quis ullam minus, perferendis dolore ipsam ut dolor ipsa quo commodi nulla? Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet quod fugit incidunt, possimus ea voluptatibus. Error tenetur quis ullam minus, perferendis dolore ipsam ut dolor ipsa quo commodi nulla? Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet quod fugit incidunt, possimus ea voluptatibus. Error tenetur quis ullam minus, perferendis dolore ipsam ut dolor ipsa quo commodi nulla?
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet quod fugit incidunt, possimus ea voluptatibus. Error tenetur quis ullam minus, perferendis dolore ipsam ut dolor ipsa quo commodi nulla? Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet quod fugit incidunt, possimus ea voluptatibus. Error tenetur quis ullam minus, perferendis dolore ipsam ut dolor ipsa quo commodi nulla? Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet quod fugit incidunt, possimus ea voluptatibus. Error tenetur quis ullam minus, perferendis dolore ipsam ut dolor ipsa quo commodi nulla?
    </p>
  </div>,
]
`;
